### PR TITLE
UIIN-2594: Add Set record for deletion option in Actions menu with Confirmation modal

### PR DESCRIPTION
**Pull request by OleksandrHladchenko1**
## Purpose

* Users should be able to set records for deletion so that records that are outdated or entered by mistake do not pollute the inventory.
* When invoked, a Confirmation Modal should display.

## Approach

* Display "Ser record for deletion" action item for **non-consortial** tenant when the user has "Inventory: Set records for deletion and staff suppress" permission.
* Display "Ser record for deletion" action item for **central** tenant when the instance is shared and the user has "Inventory: Set records for deletion and staff suppress" permission assigned at the CENTRAL tenant.
* Display "Ser record for deletion" action item for **member** tenant when the instance is shared and the user has "Inventory: Set records for deletion and staff suppress" permission assigned at the CENTRAL tenant.
OR
* When the instance is local and the user has "Inventory: Set records for deletion and staff suppress" permission assigned at the MEMBER tenant.

* DO NOT display "Ser record for deletion" action item when
   - Source=MARC
   - Suppress from discovery flag is set to 'true'
   - deleted set to true
   - leaderRecordStatus is set to 'd'
OR
   - Staff suppressed flag is set to 'true'
   - Suppress from discovery flag is set to 'true'

`onConfirm` action will be implemented in [UIIN-2595](https://issues.folio.org/browse/UIIN-2595)

## Refs

[UIIN-2594](https://issues.folio.org/browse/UIIN-2594)

## Screenshots

https://github.com/folio-org/ui-inventory/assets/85172747/de2dbdb5-0f72-450a-8f60-034ba1255d15

### CODE REVIEW

File: src/ViewInstance.js [hunk 1]
```diff
@@ -657,6 +663,19 @@ class ViewInstance extends React.Component {
...
+    const hasSetForDeletionPermission = stripes.hasPerm(setForDeletionAndSuppressPerm);
+    const canCentralTenantSetForDeletion = hasSetForDeletionPermission && isShared;
+    const canMemberTenantSetForDeletion = (isShared && this.hasCentralTenantPerm(setForDeletionAndSuppressPerm)) || (!isShared && hasSetForDeletionPermission);
+    const canSetForDeletion = hasSetForDeletionPermission || canCentralTenantSetForDeletion || canMemberTenantSetForDeletion;
+
+    const isRecordSuppressed = instance.discoverySuppress && instance?.staffSuppress;
+    const isRecordSetForDeletion = isSourceMARC
+      && instance.discoverySuppress
+      && instance?.deleted
+      && instance?.leaderRecordStatus === 'd';
```
Comment (BogdanDenis) [inline src/ViewInstance.js:676]: Please make `'d'` a constant with a descriptive name

**Review by BogdanDenis — APPROVED** [review]

File: src/ViewInstance.js [hunk 2]
```diff
@@ -657,6 +664,19 @@ class ViewInstance extends React.Component {
     const canCreateOrder = !checkIfUserInCentralTenant(stripes) && stripes.hasInterface('orders') && stripes.hasPerm('ui-inventory.instance.createOrder');
     const canReorder = stripes.hasPerm('ui-requests.reorderQueue');
     const canExportMarc = stripes.hasPerm('ui-data-export.app.enabled');
+
+    const hasSetForDeletionPermission = stripes.hasPerm(setForDeletionAndSuppressPerm);
+    const canCentralTenantSetForDeletion = hasSetForDeletionPermission && isShared;
```
Comment (mariia-aloshyna) [inline src/ViewInstance.js:669]: But all instances are shared on the central tenant, aren't they? Then why do we need to check for it? Maybe let's change for: `isCentralTenant && hasSetForDeletionPermission`?

File: src/ViewInstance.js [hunk 3]
```diff
@@ -657,6 +664,19 @@ class ViewInstance extends React.Component {
...
+    const canMemberTenantSetForDeletion = (isShared && this.hasCentralTenantPerm(setForDeletionAndSuppressPerm)) || (!isShared && hasSetForDeletionPermission);
+    const canSetForDeletion = hasSetForDeletionPermission || canCentralTenantSetForDeletion || canMemberTenantSetForDeletion;
+
+    const isRecordSuppressed = instance.discoverySuppress && instance?.staffSuppress;
```
Comment (mariia-aloshyna) [inline src/ViewInstance.js:673]: Why do you have a check for `instance` for `staffSuppress` but do not have for `discoverySuppress`?
```suggestion
    const isRecordSuppressed = instance?.discoverySuppress && instance?.staffSuppress;
```
or
```suggestion
    const isRecordSuppressed = instance.discoverySuppress && instance.staffSuppress;
```

File: src/ViewInstance.js [hunk 4]
```diff
@@ -657,6 +664,19 @@ class ViewInstance extends React.Component {
...
+    const isRecordSetForDeletion = isSourceMARC
+      && instance.discoverySuppress
```
Comment (mariia-aloshyna) [inline src/ViewInstance.js:675]: the same here

File: src/ViewInstance.js [hunk 5]
```diff
@@ -1079,6 +1107,16 @@ class ViewInstance extends React.Component {
...
             />
 
+            <ConfirmationModal
+              open={this.state.isSetForDeletionModalOpen}
+              heading={<FormattedMessage id="ui-inventory.setForDeletion.modal.header" />}
+              message={<FormattedMessage id="ui-inventory.setForDeletion.modal.message" values={{ instanceTitle: instance?.title }} />}
+              confirmLabel={<FormattedMessage id="ui-inventory.confirm" />}
+              onCancel={this.handleCloseSetForDeletionModal}
+              // Implement setting record for deletion in scope of UIIN-2595
+              // onConfirm={noop}
```
Comment (mariia-aloshyna) [inline src/ViewInstance.js:1117]:
```suggestion
              onConfirm={noop}
```

File: src/ViewInstance.test.js [hunk 6]
```diff
@@ -999,6 +999,115 @@ describe('ViewInstance', () => {
...
+            deleted: true,
+          }));
+
+          renderViewInstance();
+
+          expect(screen.queryByText('button', { name: 'Set record for deletion' })).not.toBeInTheDocument();
+        });
+      });
+
+      describe('when user is in central tenant and instance is shared', () => {
```
Comment (mariia-aloshyna) [inline src/ViewInstance.test.js:1058]: All instances are shared in central tenant

**Pull request merged**